src/components/todos/TodoItem.js

```diff
@@ -4,10 +4,45 @@ import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faTrashAlt } from '@fortawesome/free-solid-svg-icons';
 import './TodoItem.scss';
 import PropTypes from 'prop-types';
+import _ from 'lodash';
 
 class TodoItem extends Component {
-  state = {
-    isClicked: false,
+
+  constructor(props){
+    super(props);
+    this.state = {
+      isClicked: false,
+      todoTitle: props.todo.todoItem.title,
+    }
+  }
+
+  componentDidMount() {
+    this.updateTodoTitle();
+
+  }
+
+  updateTodoTitle = () => {
+    const url = new URL('http://localhost:5000/api/todos/edit');
+    // Adding parameters to url
+    url.searchParams.append('id', this.props.todo._id);
+    url.searchParams.append('todoTitle', this.state.todoTitle);
+
+    fetch(url, {
+      method: 'PUT'
+    })
+      .then(res => res.json())
+      .then(response => {
+        console.log(response);
+      })
+      .catch(error => {
+        console.log(error);
+      });
+  }
+
+  onTodoChange(value) {
```

Comment:
You should add a debounce function inside this handler, that calls updateTodoTitle(), that way the api call is made a few moments after a user stops typing to update the value. It would look something like this:

_.debounce(this.updateTodoTitle, 300))

That would fire a call to the api 300 ms after the user stops typing, this effectively reduces unnecessary calls to the api every time the user types a character.